pipelines/rbsa/tasks/apply_devicemap.py

```diff
@@ -0,0 +1,193 @@
+            
+            # generate mapping
+            column_map = {k: [] for k in enduses}
+
+            for column in list(df): 
+                col_name = column.split()[0]
+
+                if (col_name == 'Hours') | (column[:10] == 'Fahrenheit'):
+                    continue  
+                elif rbsa_dict[col_name]['unit'] == 'kWh':    
+                    column_map[rbsa_dict[col_name]['enduse']].append(column)
+            
+            # remove unused
+            column_map.pop('', None)
+            column_map.pop('ignore', None)
+            
+            for enduse in column_map.keys():
+                df[enduse] = df[column_map[enduse]].sum(axis=1)
+                
+            df = df[list(column_map.keys())]
+            
+            if initialize_master:
+                master_y1 = df.copy(deep=True)
+                initialize_master = False
+            else:
+                master_y1 = pd.concat([master_y1, df])
+
+        initialize_master = True
+
+        for file in self.Y2_files:
+            logger.info(f'Cleaning {file}')
+            # clean and resample
+            df = data_map[file]
+            df['time'] = pd.to_datetime(df['time'], format='%d%b%y:%H:%M:%S')
+            df = df.fillna(0)
+            df['siteid'] = df['siteid'].astype(str)  
+            df = df.groupby('siteid').resample('60T', on='time').sum()
+
+            # generate mapping
+            column_map = {k: [] for k in enduses}
+
+            for column in list(df):
+                try:
+                    if rbsa_dict[column]['unit'] == 'kWh':    
+                        column_map[rbsa_dict[column]['enduse']].append(column)
+                except:
+                    continue # unused columns
+                    
+            # remove unused
+            column_map.pop('', None)
+            column_map.pop('ignore', None)
+
+            for enduse in column_map.keys():
+                df[enduse] = df[column_map[enduse]].sum(axis=1)
+                
+            df = df[list(column_map.keys())]
+            
+            if initialize_master:
+                master_y2 = df.copy(deep=True)
+                initialize_master = False
+            else:
+                master_y2 = pd.concat([master_y2, df])
+
+        master_df = pd.concat([master_y1, master_y2])
+        master_df = master_df.sort_values(by='siteid')
+        master_df = master_df.reset_index(level=[0,1])
+        master_df = master_df.groupby('siteid').resample('60T', on='time').sum()
+
+        for enduse in self.enduses_needed:
+            if enduse not in master_df.columns:
+                master_df[enduse] = 0
+                logger.info(f'Adding {enduse} column with values set to 0.')
+
+        for column in self.unneded_columns:
+            if column in master_df.columns:
+                master_df = master_df.drop([column], axis=1)
+                logger.info(f'Removing column {column}.')
```

Comment:
Wouldn't you be able to drop multiple columns at once by just doing something like: `master_df.drop(self.unneded_columns)`? Would save the whole loop

Reply:
fixed